tt_metal/impl/device/device.cpp

```diff
@@ -3229,6 +3229,15 @@ void Device::set_worker_mode(const WorkExecutorMode& mode) {
 void Device::enable_async(bool enable) {
```

Comment:
(nit)(not a matter of this PR)

Please consider renaming api to set_async_enabled(true/false).
Methods with name like `enable_async`(based on name) are not expected to accept a parameter.
This is very small, but wanted to bring to your attention anyway.

Reply:
Thanks, will update in a future PR.